lib/Windowing/InViewList.js

```diff
@@ -1,17 +1,58 @@
 import React, { useContext } from 'react';
-import { arrayOf, func, shape } from 'prop-types';
+import { arrayOf, func, shape, number } from 'prop-types';
 import { WindowingContext } from './WindowingProvider';
 
-function InViewList({ items, children }) {
+const getLengthOfIndexesToRender = (
+  renderIndexes,
+  acceptedRange,
+  totalLength
+) => {
+  const endOfIndexesToRender = renderIndexes.offset + renderIndexes.length;
+  const endOfAcceptedRangeOfIndexes =
+    acceptedRange.offset + acceptedRange.length;
+
+  if (endOfIndexesToRender === endOfAcceptedRangeOfIndexes) {
+    return endOfIndexesToRender;
+  }
+
+  if (endOfIndexesToRender > endOfAcceptedRangeOfIndexes) {
+    return totalLength;
+  }
+
+  if (endOfIndexesToRender < endOfAcceptedRangeOfIndexes) {
+    return endOfIndexesToRender - renderIndexes.offset;
+  }
+
+  return -1;
```

Comment:
Have you guys seen an edge case where we need this final return?

If not, I would say get rid of it and have line 23 as the final return, without the if statement. Realistically a number can only be bigger, smaller or the same as another number. There aren't any other edge cases.